src/libs/humanizer/modules/Uniswap/utils.ts

```diff
@@ -29,3 +29,59 @@ export const joinWithAndLabel = (
 ): HumanizerVisualization[] => {
   return humanizations.reduce((acc, arr) => [...acc, ...arr, getLabel('and')], []).slice(0, -1)
 }
+
+export const uniReduce = (calls: HumanizerVisualization[][]): HumanizerVisualization[] => {
+  const reduced: HumanizerVisualization[][] = []
+  for (let i = 0; i < calls.length; i++) {
+    if (i === calls.length - 1) {
+      reduced.push(calls[i])
+      // eslint-disable-next-line no-continue
+      continue
+    }
+
```

Comment:
all of this is incomprehensible, can you think of a better way to do it? I also think that it can be kind of more generalized

also consider that in some cases the wrap won't be RIGHT before the swap (I can't think of such cases but it's theoretically possible)